### ISSUE TITLE
Pokemon Sacred Gold + ICV

### ISSUE THREAD

**Issue by NomadBal** [TheGag96/individual-color-variation#9]
Is making a version of ICV that works with Pokemon Sacred Gold (it's a ROM hack made by Drayano like Renegade Platinum) in the plans, or is it feasible?

**Comment by TheGag96** [comment]
Yeah, this can be done - I'd have to manually build it. I can try to get to this soon.

**Comment by NomadBal** [comment]
That's awesome!
I don't know if it makes any difference, but there are two versions of the hack rom, 1.05 and 1.1, the difference being the presence of the fairy typing

**Comment by TheGag96** [comment]
I apologize that I procrastinated on this so long... If you're still interested, check the releases page! If it works, please close the issue. Patch this against Sacred Gold v1.1, patched from a 4780 dump of HG.

**Comment by NomadBal** [comment]
Thank you so much, this is awesome!

Unfortunately there are a couple of things that I noticed.

The Pokedex is in spanish. 
Unfortunately this is an issue with the 1.1 version that I noticed a couple of weeks after I opened this issue and it's present even after applying your patch. The creator is from a spanish speaking country and they must've used a spanish version of the game.
I didn't play this version any more after noticing this, So I don't really know if there could be even worse translation issues farther in the game. 
I really don't know if there's something that could be done about that.
And I tested your patch up to the point you get the pokedex to verify this.
Do you think it would be possible to adapt your patch to the 1.05 version of the hack? 
That's directly from Drayano, so I really don't think that there would be ANY issue.

Then I noticed that the wild pokemons "flicker" when the spawn animation loop finishes, like they're changing color only when you can interact with the game. 
Maybe that's how the patch works? Don't know, but it's noticeable.

I tried grabbing some screenshots of some wild pidgeys and sentrets and they actually change colors, but the difference it's so faint that I had to bring the screenshots to the computer to notice something different, and even then it's barely noticeable.
Maybe it's bad luck with 8 pidgeys.
If the colors are based on the stats maybe the starting pokemons stats are so low and close together that the patch can't apply any difference at all. Otherwise would it be possible to crank this effect some more?

**Comment by TheGag96** [comment]
The Spanish Pokedex is definitely goofy... I can try patching with the lesser version. 

The flashing between palettes you see is hopefully not my fault lol - I noticed a long time ago that there's a strange display quirk with the DS where there's a difference between 3D objects that use textures (e.g. the actively transforming sprites) and tiled graphics (e.g. stationary sprites in between turns). I have to hijack different places in the code to make both kinds get their palettes modified. Despite the values being the same in both cases, on real hardware and in emulators, you can see the colors change ever so slightly visually. I'm really not sure why that is. I *think* this can be seen in the original game, too.

It is totally possible to just have bad luck with the color differences. Also, less saturated a Pokemon's sprite is, the less noticeable it can be at best anyway. Also also, real hardware, especially an original DS, will make things harder to notice since the displays just aren't as vibrant as your average phone or PC monitor. I chose the range of values I did because the min and max hue shifts start to look strange not long past that. Changing it requires running the script to regenerate the table and repatching the place it goes in the ROM. (I didn't make it dynamic, being worried about performance.) After I repatch with 1.05, I encourage you to try again with the default - I've been pretty satisfied myself with it.

**Comment by TheGag96** [comment]
@NomadBal Try the v1.05 patch I just posted on the Releases page!

**Comment by NomadBal** [comment]
Thanks again! 
I've applied the patch and I will try it on some devices throughout the day
